core/src/main/java/org/apache/iceberg/rest/RESTResponse.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+/**
+ * Interface to mark a REST response
+ */
+public interface RESTResponse extends RESTMessage<RESTResponse> {
+}
```

Comment:
Initially, I had this as `RESTResponse<T> extends RESTMessage<T>`.

Arguably that's more type safe, but it required _every single_ request / response object to redeclare itself twice.

I think it's sufficient to just have type checking on the fact that it's a request or a response, as it would be unlikely to define a class with a `validate` method that returns some other `RESTRequest` or `RESTResponse`.

If we think this trade off isn't worth it, I can revert the change.

Reply:
Yeah, I see what you mean.

I think I'd probably change `validate` to return `void` instead of `T` so that we don't need a type. As you pointed out, it's annoying to add every class as a parameter to the superclass. And the benefit isn't that large. I like being able to use `return msg.validate()` but you can also use separate statements quite easily.

The drawback to the current approach is that `msg.validate()` actually returns `RESTResponse` instead of the actual type, so you still can't return the results in most cases. I would opt for the simple version. We can safely add the type paramters and return value later.

Reply:
Agreed. With the current approach, you can return `msg.validate()` only if the return type is explicitly stated as the correct subtype, and it's up to the user to specify it correctly. So it's not providing much value.

Will update to return `void` there and remove the type parameter.